env.ts

```diff
@@ -0,0 +1,56 @@
+import {
+    defineConfig,
+    overrideDefineForWebAppServe,
+    Schema,
+} from '@julr/vite-plugin-validate-env';
+
+const webAppServeEnabled = process.env.WEB_APP_SERVE_ENABLED?.toLowerCase() === 'true';
+if (webAppServeEnabled) {
+    // eslint-disable-next-line no-console
+    console.warn('Building application for web-app-serve');
+}
+const overrideDefine = webAppServeEnabled
+    ? overrideDefineForWebAppServe
+    : undefined;
+
+export default defineConfig({
+    overrideDefine,
+    validator: 'builtin',
+    schema: {
+        VITE_FIREBASE_API_KEY: Schema.string.optional(),
+        VITE_FIREBASE_AUTH_DOMAIN: Schema.string.optional(),
+        VITE_FIREBASE_DATABASE_URL: Schema.string({ format: 'url', protocol: true, tld: false }),
+        VITE_FIREBASE_PROJECT_ID: Schema.string.optional(),
+        VITE_FIREBASE_STORAGE_BUCKET: Schema.string.optional(),
+        VITE_FIREBASE_MESSAGING_SENDER_ID: Schema.string.optional(),
+        VITE_FIREBASE_APP_ID: Schema.string.optional(),
+        VITE_FIREBASE_MEASUREMENT_ID: Schema.string.optional(),
+        VITE_BASE_URL: Schema.string({ format: 'url', protocol: true, tld: false }),
+        VITE_IMPRINT_URL: Schema.string({ format: 'url', protocol: true, tld: false }),
+        VITE_APP_NAME: Schema.string.optional(),
+        VITE_APP_ATTRIBUTION_TITLE: Schema.string.optional(),
+        VITE_APP_ATTRIBUTION_URL: Schema.string({ format: 'url', protocol: true, tld: false }),
+        VITE_APP_WEBSITE_URL: Schema.string({ format: 'url', protocol: true, tld: false }),
+        VITE_APP_LOG: Schema.string.optional(),
+        VITE_PROJECTS_FALLBACK_IMAGE: Schema.string.optional(),
+        VITE_ALLOW_UNVERIFIED_USERS: Schema.string.optional(),
```

Comment:
Lets verify this if this should be a boolean type, lets also double check types for all other env variables